### PR TITLE
Ensure clean http url

### PR DESCRIPTION
**Pull request by ryokather**
# Description

This PR fixes issue 367 in the opentelemetry-python-contrib repo. A new specification was established such that http.url, a span attribute, must not contain username or password information for security purposes. This would ensure that URLs passed in the form of `https://username:password@www.example.com/` would result in a http span attribute value of `https://www.example.com/`.

This issue was fixed by implementing an additional util function in `opentelemetry-instrumentation` called `remove_url_credentials(url: str)`. This function utilizes the `yarl.URL` module to cleanly strip a username and password from a given url if present. This fix was applied to the following instrumentations: aiohttp-client, asgi, requests, tornado, urllib, and wsgi. Note that the current implementation of the urllib3 instrumentation would not process the username and password of a url so this logic was not needed.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

An additional test called `test_credential_removal` was provided in every instrumentation that sets the `HTTP_URL` attribute. This includes the instrumentations aiohttp-client, asgi, requests, tornado, urllib, urllib3, and wsgi. Tests were run using tox and python 3.8.2.

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [x] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by Aneurysm9 — COMMENTED** [review]
This seems to repeat the same few lines of implementation over and over again.  Could that be centralized in a helper function used by the instrumentation so that if the logic needs to change it only needs to change in one place?  I did this recently in the Go SDK and it was simple because we have all of our instrumentation adding this HTTP attribute through a [single helper](https://github.com/open-telemetry/opentelemetry-go/pull/1919/files#diff-60dfb811eea483e0d043d8fe520af5c7dd0dfdde9106dac2559dae11798a1949).

**Comment by ryokather** [comment]
@Aneurysm9 Great suggestion. I just noticed that there is also a util module that is used by some instrumentations which sounds like a much more reusable place to put it. I'll update my changes shortly!

File: tox.ini
Comment (eddyleelin) [inline tox.ini]: Do you know why you added yarl here, specifically? It seems like it looks different from the rest of the lines near it, so just wondering.

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py
Comment (eddyleelin) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py]: I think this should be in the same import block as the `wrapt` import based on [PEP 8 import conventions](https://www.python.org/dev/peps/pep-0008/#imports). What do you think?

**Review by eddyleelin — COMMENTED** [review]
Really nice job! Would recommend that you fix the lint issues before submitting to the upstream! You can either look through them and manually change them by looking at the checks (that's what I usually do), or I think there's a way to automatically autoformat them using lint and black/isort, which should be listed somewhere in the repos?
Good job again! 👍

File: tox.ini
Comment (dgetu) [reply tox.ini:229]: I'd also suggest installing that package on its own line.

Comment (dgetu) [reply tox.ini]: I agree. These lines seem to be there specifically for installing OTel packages. Adding external dependencies should be done through either `requirements.txt` (if it's a dev/prod dependency) or through the `deps` field for that particular test in `tox.ini` (if it's only required for the test).

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py
Comment (ryokather) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py]: You're right! Running `isort` on the file confirmed this. I was also able to fix the other lint errors using black in vscode.

File: tox.ini
Comment (ryokather) [reply tox.ini]: Sounds good. I'm still trying to find the right place where the imports are supposed to go so all the tests and docker tests succeed but I'll dig into it a little more.

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py
Comment (Aneurysm9) [inline opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py]: Should this return a string instead of a `URL`?  It seems like it works fine for now, presumably because of implicit stringification of the `URL` where it's used, but I'm not sure what impact changing the type of this value may have.

**Review by Aneurysm9 — APPROVED** [review]
Not sure this is worth holding this up for.  Perhaps a conversation to have with the upstream maintainers who will presumably be better positioned to know whether or not it's an issue.

File: opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py
Comment (ryokather) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py]: A string would make more sense however the existing implementation of one of the instrumentations (aiohttp) and the corresponding test of `url_filter` takes advantage of the `URL` format and that's why it made sense to not be destructive of that implementation.  I'll explore if returning a string instead would be possible before submitting upstream.

Comment (eddyleelin) [reply opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py]: It seems like you've changed this to return a str now, right? Just to confirm

**Review by eddyleelin — APPROVED** [review]
Nice! LGTM. It could also help to resolve the tox.ini conflicts before submitting upstream!

**Pull request closed without merging**
